hack/docker.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+# Script that builds and pushes docker images for multiple architectures. If you want to include a new architecture,
+# ensure that it is compiled in the build.sh script, then add your OS/ARCH combination to the PLATFORMS variable.
+
+PLATFORMS=linux/arm/v7,linux/arm64/v8,linux/amd64
+DOCKER_IMAGE=quay.io/jetstack/preflight
+
+# Uses the git tag or commit SHA.
+VERSION="$(bash ./hack/version.sh)"
+
+docker buildx build \
+  --platform ${PLATFORMS} \
+  --tag "${DOCKER_IMAGE}:${VERSION}" \
+  --tag "${DOCKER_IMAGE}:latest" \
```

Comment:
Let's keep an eye on this when we merge this. I am not 100% this is going to work but it is worth a shot.

Reply:
Yeah I wasn't 100% sure it was going to fix the particular pushing problem. I was finding the build process quite hard to debug and felt like a tidy up first could make this easier to fix. It'll be great if it does too

Reply:
I wonder if we can extract this and apply just this change to the existing script to test if it works?